src/components/Dropdown/component.tsx

```diff
@@ -0,0 +1,155 @@
+        } else if (e.key === 'Enter') {
+            setIsOpen(!isOpen) //close the dropdown upon selection 
+        }
+    }
+    return (
+        <div className='relative w-full max-w-xs' onKeyDown={handleKeyDown}>
+            <div
+                className={`flex items-center justify-between w-full px-4 py-2 text-left rounded-md shadow-sm focus:outline-none focus:ring-2 focus:ring-offset-2 ${disabled ? 'bg-gray-100 cursor-not-allowed' : 'cursor-pointer hover:bg-gray-50'
+                    }`}
+                onClick={() => { !disabled && setIsOpen(!isOpen) }}
+            >
+                <span className=' block truncate'>
+                    {selected.length > 0
+                        ? selected.map(s => options.find(o => o.value === s)?.label).join(',')
+                        : <span className='text-gray-400'>{placeholder}</span>}
+                </span> {/* truncate is a tailwind class ensures that if the text is too long, it will be truncated with and ellipses(...) instead of wrapping */}
+                {/* additionally the optional chaining operator (?.) accesses the label property of the found option, even if undefined. */}
+                <div className='flex items-center'>
+                    {selected.length > 0 && !disabled && (
+                        <button
+                            type='button'
+                            className='p-1 text-gray-400 hover:text-gray-500 focus:outline-none focus:text-gray-500'
+                            onClick={(e) => {
+                                // e.stopPropagation prevents click event from bubbling up to parent elements, preventing dropdown toggle
+                                e.stopPropagation();
+                                handleClear();
+                            }}
+                        >
+                            <X size={16} /> {/* This is a Lucide React component usage */}
+                        </button>
+                    )}
+                    <ChevronDown
+                        size={20}
+                        className={`ml-2 text-green-400 transition-transform duration-200 ${isOpen ? 'rotate-180' : ''}`}
+                    />
+                </div>
+            </div>
+
+            {isOpen && (
+                <div
+                    ref={dropdownRef}
+                    className='absolute z-10'
+                    style={{ maxHeight: `${dropdownHeight}px` }}
+                >
+                    <div>
+                        <input
+                            type='text'
+                            placeholder="Search..."
+                            value={searchTerm}
+                            onChange={(e) => setSearchterm(e.target.value)}
+                            onClick={(e) => e.stopPropagation()}
+                        />
+                    </div>
+                    <ul className='py-1 overflow-auto text-base leading-6 max-h-60'>
```

Comment:
I think this file would benefit from pulling apart the dropdown a bit into a search bar, option list, and options. That would hugely benefit readability.